pkg/tsdb/graphite/graphite.go

```diff
@@ -10,51 +10,141 @@ import (
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin"
 	"github.com/grafana/grafana/pkg/setting"
 	"github.com/opentracing/opentracing-go"
 )
 
-type GraphiteExecutor struct {
+const (
+	dsName = "grafana-graphite-datasource"
+)
+
+var (
+	glog = log.New("tsdb.graphite")
+)
+
+type graphiteExecutor struct {
 	httpClientProvider httpclient.Provider
+	logsService        log.Logger
+	im                 instancemgmt.InstanceManager
+	cfg                *setting.Cfg
 }
 
-// nolint:staticcheck // plugins.DataPlugin deprecated
-func New(httpClientProvider httpclient.Provider) func(*models.DataSource) (plugins.DataPlugin, error) {
-	// nolint:staticcheck // plugins.DataPlugin deprecated
-	return func(dsInfo *models.DataSource) (plugins.DataPlugin, error) {
-		return &GraphiteExecutor{
-			httpClientProvider: httpClientProvider,
-		}, nil
-	}
+type GraphiteService struct {
+	logger               log.Logger
+	BackendPluginManager backendplugin.Manager `inject:""`
+	Cfg                  *setting.Cfg          `inject:""`
+	HTTPClientProvider   httpclient.Provider   `inject:""`
 }
 
-var glog = log.New("tsdb.graphite")
+type datasourceInfo struct {
+	HTTPClient              *http.Client
+	URL                     string
+	JSONData                map[string]interface{}
+	DecryptedSecureJSONData map[string]string
+	BasicAuthEnabled        bool
+	BasicAuthUser           string
+	Id                      int64
+}
 
-//nolint: staticcheck // plugins.DataQuery deprecated
-func (e *GraphiteExecutor) DataQuery(ctx context.Context, dsInfo *models.DataSource, tsdbQuery plugins.DataQuery) (
-	plugins.DataResponse, error) {
-	// This logic is used when called from Dashboard Alerting.
-	from := "-" + formatTimeRange(tsdbQuery.TimeRange.From)
-	until := formatTimeRange(tsdbQuery.TimeRange.To)
+func NewInstanceSettings(httpClientProvider httpclient.Provider) datasource.InstanceFactoryFunc {
+	return func(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		opts, err := settings.HTTPClientOptions()
+		if err != nil {
+			return nil, err
+		}
 
-	// This logic is used when called through server side expressions.
-	if isTimeRangeNumeric(*tsdbQuery.TimeRange) {
-		var err error
-		from, until, err = epochMStoGraphiteTime(*tsdbQuery.TimeRange)
+		client, err := httpClientProvider.New(opts)
 		if err != nil {
-			return plugins.DataResponse{}, err
+			return nil, err
 		}
+
+		jsonData := map[string]interface{}{}
+		err = json.Unmarshal(settings.JSONData, &jsonData)
+		if err != nil {
+			return nil, fmt.Errorf("error reading settings: %w", err)
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("error reading settings: %w", err)
+		}
+		model := datasourceInfo{
+			BasicAuthEnabled:        settings.BasicAuthEnabled,
+			HTTPClient:              client,
+			URL:                     settings.URL,
+			JSONData:                jsonData,
+			DecryptedSecureJSONData: settings.DecryptedSecureJSONData,
```

Comment:
Would remove these and replace with a field for BasicAuthPassword.

Reply:
you are right :)